File: src/main/java/cz/muni/fi/pa039/widgets/PointWidget.java
package cz.muni.fi.pa039.widgets;

import cz.muni.fi.pa039.Point;

public class PointWidget extends AbstractWidget {

    public Point point;
    public int radius;

    public PointWidget(Point point, int radius) {
        this.point = point;
        this.radius = radius;
    }

    @Override
    public int getStroke() {
        return 0;
    }

    @Override
    public int getFill() {
        return 0;
    }

    @Override
    public int getId() {
        return -1;
    }

    @Override
    public boolean pointIsInside(int x, int y) {
        return Math.sqrt(Math.pow(x - this.point.x, 2) + Math.pow(y - this.point.y, 2)) <= radius;
    }
}

File: src/cz/muni/fi/pa039/widgets/PointWidget.kt
package cz.muni.fi.pa039.widgets

import cz.muni.fi.pa039.Point

class PointWidget(var point: Point, var radius: Int) : AbstractWidget {
    override var isSelected: Boolean = false
    override val fill = 0
    override val id = -1
    override fun pointIsInside(x: Int, y: Int): Boolean {
        return Math.sqrt(Math.pow((x - this.point.x).toDouble(), 2.0) + Math.pow((y - this.point.y).toDouble(), 2.0)) <= radius
    }
}